packages/router-bridge/src/sagaRouter/sagaRouter.ts

```diff
@@ -0,0 +1,174 @@
+ *	// route configuration, a url fragment match with a generator
+ *	const routes = {
+ *		"/datasets/add": function* addDataset() {
+ *			yield take(CANCEL_ACTION);
+ *			yield put({
+ *				type: REDIRECT_ADD_DATASET_CANCEL,
+ *				cmf: {
+ *					routerReplace: "/datasets"
+ *				}
+ *			});
+ *		},
+ *		"/connections/:datastoreId/edit/add-dataset": function* addDataset({
+ *			datastoreId
+ *		}) {
+ *			yield take(CANCEL_ACTION);
+ *			yield put({
+ *				type: REDIRECT_CONNECTION_ADD_DATASET_CANCEL,
+ *				cmf: {
+ *					routerReplace: `/connections/${datastoreId}/edit`
+ *				}
+ *			});
+ *		}
+ *	};
+ *	// router saga is spawned and given router history, and route configuration
+ *	yield spawn(routerSaga, history, routes);
+ */
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { History } from 'history';
+import isEqual from 'lodash/isEqual';
+import { Task } from 'redux-saga';
+import { spawn, take, cancel } from 'redux-saga/effects';
+
+import cmf from '@talend/react-cmf';
+
+export interface Location {
+	pathname: string;
+}
+
+export interface RunningTasks {
+	[key: string]: MaybeSaga;
+}
+
+export interface RouteParams {
+	[key: string]: number;
+}
+
+export interface Match {
+	path: string;
+	url: string;
+	isExact: boolean;
+	params: RouteParams;
+}
+
+export interface MaybeSaga {
+	saga: Task;
+	match: Match;
+}
+
+export interface RouteSaga {
+	saga: MaybeSaga;
+	params: RouteParams;
+	runOnExactMatch: boolean;
+	restartOnRouteChange: boolean;
+}
+
+export interface RoutesConfig {
+	[key: string]: RouteSaga;
+}
+
+/**
+ * Determine if a saga should be restarted with the following rules :
+ */
+function shouldStartSaga(maybeSaga: MaybeSaga, match: Match, routeSaga: RouteSaga) {
+	if (match) {
+		if (!maybeSaga || (maybeSaga && maybeSaga.saga && !maybeSaga.saga.isRunning())) {
+			if (routeSaga.runOnExactMatch === true) {
+				return match.isExact;
+			}
+			return true;
+		}
+	}
+	return false;
+}
+
+/**
+ * Determine if a saga should be canceled with the following rules :
+ */
+function shouldCancelSaga(maybeSaga: MaybeSaga, match: Match, routeSaga: RouteSaga) {
+	if (maybeSaga && maybeSaga.saga.isRunning()) {
+		if (!match || routeSaga.runOnExactMatch === true) {
+			return true;
+		}
+	}
+	return false;
+}
+
+/**
+ * Determine if a saga should be restarted with the following rules:
+ */
+function shouldRestartSaga(maybeSaga: MaybeSaga, match: Match, routeSaga: RouteSaga) {
+	if (match) {
+		if (maybeSaga) {
+			if (
+				routeSaga.restartOnRouteChange === true ||
+				!isEqual(maybeSaga.match.params, match.params)
+			) {
+				return true;
+			}
+		}
+	}
+	return false;
+}
+
+/**
+ * for a route a list of running saga and current location return a
+ * match object and a saga
+ */
+function parseSagaState(routeFragment: string, sagas: RunningTasks, currentLocation: Location) {
+	return {
+		match: cmf.router.matchPath(currentLocation.pathname, { path: routeFragment }),
```

Comment:
Shouldn't we use [RR v5 > mathPath](https://v5.reactrouter.com/web/api/matchPath) instead, and adapt the code that use the match result if needed ?